### ISSUE TITLE
plot displayed in log console instead of in notebook

### ISSUE THREAD

**Issue by rdelubi** [jupyter-widgets/ipywidgets#2892]
I have a notebook with two cells.

1st cell:
```
import pandas as pd 
import datetime
import time
import plotly.express as px
import plotly.graph_objs as go
from plotly.subplots import make_subplots
from IPython.core.display import display, HTML, clear_output
from ipytree import Node, Tree
import ipywidgets as widgets
from ipywidgets import HBox, VBox, interact, HTML
import base64
import numpy as np
```

2nd cell:
```
tree = Tree()

class factor_class(Node):
    model = 0

model_node = Node('Test Model 1', opened = False)
tree.add_node(model_node)
new_factor = factor_class('Factor 1')
new_factor.model = 1
model_node.add_node(new_factor)

model_node = Node('Test Model 2', opened = False)
tree.add_node(model_node)
new_factor = factor_class('Factor 1')
new_factor.model = 1
model_node.add_node(new_factor)

plot_area = HBox()

dropdown_cummulative = widgets.Dropdown(options = ['Cumulative Sum', 'Cumulative Product'], 
                                   description="Method:")

datepicker_start = widgets.DatePicker( 
    description='Start Date',
    disabled=False,
    value=datetime.datetime.now() - datetime.timedelta(30))

datepicker_end = widgets.DatePicker( 
    description='End Date',
    disabled=False,
    value=datetime.datetime.now())

def plot_handler(event):
    x = np.arange(10)
    fig = go.Figure(data=go.Scatter(x=x, y=x**2))
    fig.show()
    display(fig)

button_plot = widgets.Button(description="Plot")
button_plot.on_click(plot_handler)

column1 = widgets.VBox([
    datepicker_start,
    datepicker_end,
    dropdown_cummulative,
    button_plot
])

list_widgets  = [widgets.HBox([column1, tree])]
accordian = widgets.Accordion(children=list_widgets)
display(accordian)
```

After execution of 2nd cell, when I click the "Plot" button, the plot gets display in the log console (actual outcome) as opposed to displayed in notebook (expected outcome).

Any thoughts on why it's happening like that?

My environment:

```
Jupyterlab: 2.1.2

sh-4.2$ jupyter labextension list
JupyterLab v2.1.2
Known labextensions:
   app dir: /opt/app-root/share/jupyter/lab
        @j123npm/jupyterlab-dash v0.1.0-alpha.4  enabled  OK
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/debugger v0.3.0-alpha.1  enabled  OK
        @jupyterlab/git v0.20.0  enabled  OK
        @jupyterlab/latex v2.0.0  enabled  OK
        ipytree v0.1.8  enabled  OK
        jupyterlab-plotly v4.8.0  enabled  OK
        jupyterlab-sql v0.3.3  enabled   X
        nbdime-jupyterlab v2.0.0  enabled  OK
        plotlywidget v4.8.0  enabled  OK
        qgrid2 v1.1.3  enabled  OK

   The following extension are outdated:
        jupyterlab-sql
        
   Consider running "jupyter labextension update --all" to check for updates.

sh-4.2$ 


sh-4.2$ pip list |grep widget
ipywidgets             7.5.0
widgetsnbextension     3.5.1
WARNING: You are using pip version 20.1; however, version 20.1.1 is available.
You should consider upgrading via the '/opt/app-root/bin/python3 -m pip install --upgrade pip' command.
sh-4.2$ 
```

Note that if I switch from jupyterlab UI to classic UI and rerun the same notebook, the plot gets displayed in the notebook as expected.

**Comment by yitzisnow** [comment]
I am encountering the same issue, did you ever find a solution?

**Comment by vidartf** [comment]
tl;dr: I would try using an `Output` widget to control where the widget gets displayed. 

Some background explanation: When calling `plot_handler` in reaction to a widget event (button click), it isn't immediately apparent where widget outputs should be sent (there is no cell that is currently executing in most cases). Output widgets where created for this purpose, so that you can explicitly declare where widgets should go in such cases.

**Comment by starovoitovs** [comment]
Make sure you update all jupyter related packages including `ipypml` and `ipywidgets`

**Comment by scottwsides** [comment]
> Make sure you update all jupyter related packages including `ipypml` and `ipywidgets`

what versions should these be?

**Comment by starovoitovs** [comment]
I upgraded to the newest versions (newest at the time of my reply).

**Comment by scottwsides** [comment]
ipypml is part of jupyterlab? I've never had to install that before
